Source/Adobe.Target.Client/OnDevice/DecisioningDetailsExecutor.cs

```diff
@@ -0,0 +1,323 @@
+                },
+                _ =>
+                {
+                    if (prefetchResponse == null)
+                    {
+                        return false;
+                    }
+
+                    var responseView = new View(consequenceName, null, consequenceOptions, consequenceMetrics);
+                    var views = prefetchResponse.Views;
+                    if (views == null)
+                    {
+                        views = new List<View>();
+                        prefetchResponse.Views = views;
+                    }
+
+                    var foundView = views.Find(view => view.Name == responseView.Name);
+                    if (foundView == null)
+                    {
+                        views.Add(responseView);
+                        return true;
+                    }
+
+                    if (responseView.Options != null)
+                    {
+                        foundView.Options ??= new List<Option>();
+                        foundView.Options.AddRange(responseView.Options);
+                    }
+
+                    if (responseView.Metrics != null)
+                    {
+                        foundView.Metrics ??= new List<Metric>();
+                        foundView.Metrics.AddRange(responseView.Metrics);
+                    }
+
+                    return true;
+                });
+        }
+
+        private static Notification CreateNotification(RequestDetailsUnion details, List<Option> options, string globalMbox)
+        {
+            var id = Guid.NewGuid().ToString();
+            var impressionId = Guid.NewGuid().ToString();
+            var timestamp = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
+            var tokens = options == null ? new List<string>() : options.Select(option => option.EventToken).ToList();
+            var notification = new Notification(id: id, type: MetricType.Display, impressionId: impressionId, timestamp: timestamp, tokens: tokens);
+            return details.Match(
+                _ =>
+                {
+                    notification.Mbox = new NotificationMbox(globalMbox);
+                    return notification;
+                }, mboxRequest =>
+                {
+                    notification.Mbox = new NotificationMbox(mboxRequest.Name);
+                    return notification;
+                }, viewRequest =>
+                {
+                    notification.View = new NotificationView(viewRequest.Name, viewRequest.Key);
+                    return notification;
+                });
+        }
+
+        private static List<Metric> GetMetrics(object metricsObject)
+        {
+            return metricsObject is not JObject metricsJObject ? null : metricsJObject.ToObject<List<Metric>>();
+        }
+
+        private static void UnhandledResponse(RequestDetailsUnion details, PrefetchResponse prefetchResponse, ExecuteResponse executeResponse)
+        {
+            _ = details.Match<object>(
```

Comment:
Same here could we please, matchMbox, matchViews etc having these long `Match` method calls is extremely hard to read.

Reply:
Fixed